### DISCUSSION TITLE
Welcome to CAF Discussions!

### DISCUSSION THREAD

**Discussion by Neverlord** [actor-framework/actor-framework#1184] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Super-long** [comment]
Hello everyone, When I compile and execute the following code, an error occurs
`
/*g++ -g test1.cpp -std=c++17 -lcaf_core -o test*/
#include <iostream>
#include <string>
#include <caf/all.hpp>
#include <caf/io/all.hpp>

caf::behavior msgHandle(caf::event_based_actor *self)
{
    return{
        [=](std::string &what)
        {
            std::cout<<"string type msg is :"<<what<<std::endl;
        },
        [=](int &value)
        {
            std::cout<<"int type msg is:"<<value<<std::endl;
        }
    };   
}

int main()
{
    std::cout<<"main.cpp"<<std::endl;
    caf::actor_system_config cfg;
    caf::actor_system system(cfg);
    auto a1 = system.spawn(msgHandle);
    caf::anon_send(a1,"String message type");
    caf::anon_send(a1,123456789);

    return 0;
}
`
`
main.cpp
[FATAL] /home/lizhaolong/Desktop/Exercise/ActorFramework/actor-framework/libcaf_core/src/actor_system.cpp:315: critical error: 'actor_system created without calling caf::init_global_meta_objects<>() before'
[1]    46383 abort (core dumped)  ./main
`
If I use C++11, it will directly compile error.

**Reply by Neverlord** [reply]
I don't think this really belongs here. 😉
Our general advise is simply to use `CAF_MAIN` as shown in the examples. If you can't do that, you'll have to initialize global state manually. For the core and then for each module you're using. When using the I/O module and adding some types in `my_app`, then that's:

```c++
  caf::init_global_meta_objects<caf::id_block::my_app>();
  caf::io::middleman::init_global_meta_objects();
  caf::core::init_global_meta_objects();
```

Call these before calling *any* CAF function. Ideally right at the top of `main`.
